backend/src/Edna/Web/Types.hs

```diff
@@ -17,4 +43,180 @@ data ExperimentalMeasurement = ExperimentalMeasurement
   , emOutlier :: Bool
   } deriving stock (Generic, Show, Eq)
 
-deriveJSON ednaAesonWebOptions ''ExperimentalMeasurement
+----------------
+-- General types
+----------------
+
+-- | A simple wrapper over 'Word'. At the data layer, we identify all entities
+-- with numeric IDs and this data type corresponds to such an ID.
+-- It has a phantom parameter type which the type of the object
+-- identified by this ID.
+newtype SqlId t = SqlId
+  { unSqlId :: Word
+  } deriving stock (Generic, Show)
+    deriving newtype (FromHttpApiData, FromJSON, ToJSON, ToSchema)
+
+-- | This data type is useful when you want to return something with its ID.
+data WithId t = WithId
+  { wiId :: SqlId t
+  , wItem :: t
+  } deriving stock (Generic, Show)
+
+-- | This data type is used when you want to return something with its ID and
+-- some additional data that was not submitted by end users, but is maintained
+-- by the application.
+data WithExtra t e = WithExtra
+  { weId :: SqlId t
+  , weItem :: t
+  , weExtra :: e
+  } deriving stock (Generic, Show)
+
+-- | A stub to specify the sorting order, most likely will be replaced with
+-- @servant-util@.
+data StubSortBy =
+    SortByName
+  | SortBySomething
+
+instance FromHttpApiData StubSortBy where
+  parseQueryParam = \case
+    "name" -> pure SortByName
+    "something" -> pure SortBySomething
+    x -> Left $ "unknown sorting order: " <> x
+
+----------------
+-- Entities
+----------------
+
+-- | Input data submitted along with uploaded file.
+data FileUploadInput = FileUploadInput
+  { fuiProject :: SqlId Project
+  -- ^ ID of the project the file belongs to.
+  , fuiTestMethodology :: SqlId TestMethodology
```

Comment:
We also have description on the design.

Reply:
And there is also a lack of description in DB schema

Reply:
I think the description in the design is useless, I don't understand why it's there, we never show it. I already [wrote](https://serokell.slack.com/archives/C01MH299XEU/p1615458841001300) about it a few days ago. I hope it will be removed soon, I'll remind about it.